app/controllers/homepage_controller.rb

```diff
@@ -134,7 +134,7 @@ def find_listings(params, listings_per_page, filter_params, includes)
     search = {
       # Add listing_id
       categories: filter_params[:categories],
-      listing_shape_id: Maybe(filter_params)[:listing_shape].or_else(nil),
+      listing_shape_id: filter_params[:listing_shape] ? [filter_params[:listing_shape]] : nil,
```

Comment:
I think that instead of `Array` or `nil`, the code would be simpler if there were Array of one and Array or none. That way all the code that uses this can rely on the fact that it's an array, and they don't need to build extra `nil` checks.

Btw, `Array` method may be useful for this. It returns empty array, if you pass it a `nil`:

```ruby
2.1.1 :012 > Array(nil)
 => []
2.1.1 :013 > Array(123)
 => [123]
2.1.1 :014 >
```

So the code would be simply:

```ruby
listing_shape_ids: Array(filter_params[:listing_shape]),
```